File: Java/AbstractFactory3/src/org/TheRayCode/demo/ColorFactory.java
package org.TheRayCode.demo;

public class ColorFactory extends AbstractFactory {

    @Override
    Shape getShape(String color) {

        return null;
    }

    @Override
    Color getColor(String color) {
        if(color==null){
            return null;
        }
        if (color.equalsIgnoreCase("RED")){
            return new Red();
        } else if (color.equalsIgnoreCase("BLUE")){
            return new Blue();
        } else if (color.equalsIgnoreCase("GREEN")){
            return new Green();
        }
        return null;
    }


}

File: Csharp/Creational/AbstractFactory/Demo/Demo/ColorFactory.cs
namespace Demo
{
    class ColorFactory
    {
        public FactoryDataItem GetData(int type)
        {
            FactoryDataItem factoryDataItem = null;
            switch ((ColorType)type)
            {
                case ColorType.Red:
                    factoryDataItem = new FactoryDataItem(new Red());
                    break;
                case ColorType.Blue:
                    factoryDataItem = new FactoryDataItem(new Blue());
                    break;
                case ColorType.Green:
                    factoryDataItem = new FactoryDataItem(new Green());
                    break;
            }
            
            return factoryDataItem;
        }
    }
}